### DISCUSSION TITLE
Issues updating packages: reanimated 2.9.1 -> 3.02, expo 46 -> 48

### DISCUSSION THREAD

**Discussion by benrbowers** [nandorojo/solito#386] [Q&A] [answered]
_TL;DR: Upgrading reanimated to v3 has snowballed into having to update most of my packages. Which starter repo should I be referring to for the latest recommended package versions and babel plugins?_

I've been using solito + nativewind for a while on a cross-platform app. It's been pretty great so far, but I recently had to rewrite an animation-heavy component to use reanimated sharedValues instead of moti. I got it working on native, but then found out that it didn't work on web. It looks like reanimated doesn't officially support web until v3, and the version I had installed was 2.9.1, which is from the tailwind starter repo (https://solito.dev/tailwind). 

I started the project using that repo around December or January, so it's definitely time to update some of the packages, but it has been a pain to get everything working again with updated packages. The main two issues seem to be:
- On native: Expo 48 seems to only work with reanimated 2.14.4. How do I get it to work with **3.0.2**?
- On web: NextJs is having a lot of "You may need an appropriate loader to handle this file type." babel errors. I think this basically means that I don't have the correct plugins enabled, but I have no idea which ones.

I've tried updating my project to match the current state of the [nativewind starter repo](https://github.com/nandorojo/solito/tree/master/example-monorepos/with-tailwind), but it seems that that may already be outdated.

Am I better off referencing the main solito starter repo for the recommended package versions and plugins? Is there a better strategy I should be using to update my packages other that just looking at the current repos?

Thanks in advance! 🙌 

workspace:
 - yarn v1.22.19
 - node v16.19.0

**Comment by nandorojo** [comment, marked as answer]
- native: you need to use an expo dev client to use reanimated v3
- web: remove any babel config files you might have in the next folder. any reanimated hooks you use must have a dependency array to work properly on web. see the reanimated web docs for using web without a babel plug-in. also, remove the reanimated SWC plug-in from the next config if you have one.
- reference solito v3+ release notes to see what changes were made to upgrade the starters

**Reply by benrbowers** [reply]
This solved the majority of my problems 👍 
Basically just needed to use the expo dev client. My reanimated animation is now working on native AND web.

Only thing I had to do that's not mentioned here or in the release notes is I had to add the following plugin to my apps/expo/babel.config.js:
`@babel/plugin-proposal-export-namespace-from`
https://docs.swmansion.com/react-native-reanimated/docs/fundamentals/web-support/

Still having some issues with nextJS 13.0.0 -> 13.2, I think probably related to not transpiling the correct packages due to @expo/next-adapter v4 -> v5. For now I'm just going to coast on v4 and deal with it when it comes time to ship.
